src/plugins/auto/src/plugin.cpp

```diff
@@ -796,8 +807,9 @@ std::string Plugin::get_device_list(const ov::AnyMap& properties) const {
                         // The device is the device with default device ID (eg. GPU.0) and
                         // its wide name (eg. GPU) has been in device candidate list.
                         continue;
-                    // Add user specified device into candidate list
-                    devices_merged.push_back(device);
+                    std::string realName = parsed.get_device_name();
+                    if (is_any_dev(realName, device_list))
+                        devices_merged.push_back(device);
```

Comment:
this is a must change?

Reply:
Yes, this logic will filter out the device which is not in the registered list.
You mean we can just enable this filtering logic in the function `AUTO::parse_meta_devices()` only?
As discussed with @peterchen-intel , we implemented this logic in both function AUTO::get_device_list() and AUTO::parse_meta_devices() for double insurance.

Reply:
I found there is a test suit for` CachingTest` that needs to load network to` AUTO:mock.0,mock.1,....`.
@peterchen-intel , so should we remove this filtering logic from `AUTO::get_device_list()` and just try to catch the exception within the `AUTO::parse_meta_devices()` function?

Reply:
Have reverted the logic of `AUTO:get_device_list()`.